docs/specifications/tests/Nameserver-TP/nameserver11.md

```diff
@@ -59,35 +65,78 @@ message. The argument names are defined in the [argument list].
 
 In this section and unless otherwise specified below, the term "[EDNS Query]"
 follows the specification for DNS queries as specified in [DNS Query and Response Defaults].
-The handling of the DNS responses on the DNS queries follow, unless otherwise specified below, 
+The handling of the DNS responses on the DNS queries follow, unless otherwise specified below,
 what is specified for [EDNS Response] in the same specification.
 
 1. Create the following empty sets:
-   1. Name server IP address ("Unknown Option Code")
+   1. Name server IP address ("No Response on Unknown Option Code")
+   2. Name server IP address and [RCODE Name] ("Unexpected RCODE on Unknown Option Code")
+   3. Name server IP address ("No EDNS on Unknown Option Code")
+   4. Name server IP address ("Unexpected Answer Section on Unknown Option Code")
+   5. Name server IP address ("Unset AA on Unknown Option Code")
+   6. Name server IP address ("Returns Unknown Option Code")
 
 2. Create a [EDNS Query] with query type SOA, *Child Zone* as query name and with
+   no EDNS options of flags ("SOA Query").
+
+3. Create a [EDNS Query] with query type SOA, *Child Zone* as query name and with
    EDNS OPTION-CODE set to anything other than what is already assigned in
-   the [IANA-DNSSYSTEM-PARAMETERS] and no other EDNS options or flags ("SOA Query").
+   the [IANA-DNSSYSTEM-PARAMETERS] and no other EDNS options or flags
+   ("SOA Query with EDNS Option").
 
-3. Obtain the set of name server IP addresses using [Method4] and [Method5] 
+4. Obtain the set of name server IP addresses using [Method4] and [Method5]
    ("Name Server IP").
 
-4. For each name server in *Name Server IP* do:
+5. For each name server in *Name Server IP* do:
+
+   1. Send *SOA Query* to the name server and collect the response.
+   2. Go to next name server if at least one of the following criteria is met:
+      1. There is no DNS response from the server.
+      2. EDNS is unset in the response.
```

Comment:
This relates to a standard SOA query (no EDNS) so I don't think this step should exist. Unless maybe `is set` was intended instead?

Reply:
No, it is correct. The SOA query is with EDNS:
```
2. Create a [EDNS Query] with query type SOA, *Child Zone* as query name and with
   no EDNS options or flags ("SOA Query").
```
Should step 2 be clarified?

Reply:
I think you can leave as is